### ISSUE TITLE
Check out any other software that might be worth supporting

### ISSUE THREAD

**Issue by nanos** [nanos/FediFetcher#143]
See here for list of software: https://fedidb.org/software

**Comment by nanos** [comment]
The following are the most commonly encountered ons by myself 
- GoToSocial
- Friendica
- WordPress

**Comment by colinstu12** [comment]
I found a friendica post that seems to properly fetch when I search for it in masto web (doesn't work in Ivory however): 
https://friendica.myportal.social/display/e65e1095-2066-778e-a9b1-719897452963
Not sure why friendica is so hard to explore / seems so locked down, not sure if that's just an instance configuration thing or what.

**Comment by nanos** [comment]
Thanks @colinstu12 

I did try this. According to the docs, the API to get context should be the same as Mastodon: `GET /api/v1/statuses/:id/context`. However, when I try to get `https://friendica.myportal.social/api/v1/statuses/e65e1095-2066-778e-a9b1-719897452963/context` I only get back a `404`. So I'm not sure if I need to convert that UUID into a different ID, or if the docs are wring, or if the endpoint requires auth, and the `404` is just a masked `403`. 

All very confusing 🤔

**Comment by nanos** [comment]
OK, so I'm a little bit closer here: Looking at the source code I find a reference to a numeric ID for this post, and if I use that it works:

`https://friendica.myportal.social/api/v1/statuses/45055616/context`

But how can I convert the UUID to a numeric ID?

**Comment by edent** [comment]
For WordPress, are you looking at ActivityPub powered ones, or just regular blogs?

**Comment by nanos** [comment]
@edent Yes, ActivityPub ones. Here is a node info example response:

```json
{
  "version": "2.0",
  "software": {
    "name": "wordpress",
    "version": "6.5"
  },
  "usage": {
    "users": {
      "total": 1,
      "activeMonth": 1,
      "activeHalfyear": 1
    },
    "localPosts": 340,
    "localComments": 11
  },
  "openRegistrations": true,
  "services": {
    "inbound": [
      "atom1.0",
      "rss2.0",
      "pop3"
    ],
    "outbound": [
      "atom1.0",
      "rss2.0",
      "wordpress",
      "smtp"
    ]
  },
  "protocols": [
    "activitypub"
  ],
  "metadata": {
    "generator": {
      "name": "NodeInfo WordPress-Plugin",
      "version": "2.3.1",
      "repository": "https://github.com/pfefferle/wordpress-nodeinfo/"
    },
    "nodeName": "Symfony Station / The Payload Newsletter",
    "nodeDescription": "The Symfony Station Fediverse Account and Newsletter",
    "nodeIcon": "https://newsletter.mobileatom.net/wp-content/uploads/2023/12/cropped-Symfony-Station-No-Text-Astronaught-2C.png"
  }
}
```

**Comment by edent** [comment]
Sorry if I've misunderstood this, but I think this is the way to do it.

If you look at my blog's WebFinger - https://shkspr.mobi/blog/.well-known/webfinger?resource=acct:blog@shkspr.mobi - that points to https://shkspr.mobi/blog/@blog

If you request that with `Accept: application/activity+json`, you'll get back a standard AP document which says the outbox is https://shkspr.mobi/blog/wp-json/activitypub/1.0/actors/0/outbox

That's paginated, so should let you backfill all the contents.

**Comment by SimplyCorbett** [comment]
Any chance you can add partial support for the iceshrimp.net beta? It has mastodon compatible api but is missing features currently.

https://is.ready.social/swagger/ for a list of the api. You can select mastodon from the dropdown in the top right.

**Comment by nanos** [comment]
FediFetcher does already support ice shrimp. Is there any specific issue you’ve encountered?

**Comment by SimplyCorbett** [comment]
> FediFetcher does already support ice shrimp. Is there any specific issue you’ve encountered?

I wasn’t able to get it to work with iceshrimp.net but could with mastodon. Are you talking about iceshrimp.js? I’m talking about their new codebase on the beta/dev channel at iceshrimp.net. The software isn’t finished yet.

**Comment by nanos** [comment]
@SimplyCorbett I'm talking about the software IceShrimp, rather than about a specific server.

As of today, iceshrimp.net doesn't appear to be hosting a Fediverse instance. The `.well-known/nodeinfo` endpoint is a simple redirect to a code repo, as is `host-meta`

```
curl -i https://iceshrimp.net/.well-known/nodeinfo                                                                                                                                                                                                                                                                                                               
HTTP/2 301
alt-svc: h3=":443"; ma=2592000
location: https://iceshrimp.dev/iceshrimp/iceshrimp.net
server: Caddy
```

**Comment by colinstu12** [comment]
What about https://next.iceshrimp.dev ?

**Comment by nanos** [comment]
Thanks @colinstu12  

if they do support the mastodon api then it should be really straightforward to implement. I’ll have a look at it some time next week

**Comment by nanos** [comment]
@SimplyCorbett @colinstu12 can one of you test if #153 works correctly with Iceshrimp.NET?

**Comment by colinstu12** [comment]
> @SimplyCorbett @colinstu12 can one of you test if #153 works correctly with Iceshrimp.NET?

They appear to still be in closed beta/alpha stage. But it sounded like late 2024 or early 2025 there'd be a release. All of the Iceshrimp.js team has swapped over to the .net one it sounds like so it's good to support it, but can't quite test yet. If those iceshrimp.net folks run across this or if they're reached out to on fedi maybe they can assist further.

**Comment by nanos** [comment]
> They appear to still be in closed beta stage. 

then why did @SimplyCorbett ask for support? 🤔😁

**Comment by SimplyCorbett** [comment]
It’s open beta. They are targeting an official release by the end of 2024.

I’m asking for support because I use iceshrimp.net as my daily driver.

I will give #153 a shot, thanks!

**Comment by nanos** [comment]
Thanks  for the clear @SimplyCorbett 

if your feedback is positive I’ll be happy to merge.

**Comment by SimplyCorbett** [comment]
> Thanks for the clarification @SimplyCorbett
> 
> if your feedback is positive I’ll be happy to merge.

I didn’t run into any issues and it finished successfully, but I also dont have a lot of things setup in the config.

So it appears to work, thanks!

**Comment by nanos** [comment]
Great, thanks for the feedback @SimplyCorbett

**Comment by colinstu12** [comment]
Hello @nanos , why exactly can GoToSocial not be supported? I have a friend on a GTS instance now and I notice in my fedifetcher logs this now: unknown server api for social.bateleur.org 
I seem to be able to pull up context just fine from Masto Web UI and from Ivory client (when using both the gotosocial post link or my instance's local copy of the post too)? Why can't fedifetcher perform the same context retrieves etc? 
Thank you

**Comment by colinstu12** [comment]
> @colinstu12 see the 2nd comment in this thread: GTS requires authentication for their api, so we cannot support them. In the linked issue the maintainer explains their rationale..

Would #183 help at all with this?

**Comment by nanos** [comment]
No, sadly not. #183 means that you can run (some parts of) FediFetcher if your home instance is GoToSocial. 

But it does nothing for fetching stuff from remote instances.

**Comment by mio-19** [comment]
> [@colinstu12](https://github.com/colinstu12) see the 2nd comment in this thread: GTS requires authentication for their api, so we cannot support them. In the linked issue the maintainer explains their rationale..


The link is dead and not available on web archive. Is it on codeberg?

**Comment by nanos** [comment]
> The link is dead and not available on web archive. Is it on codeberg?

@mio-19 I haven't been able to find it, no.

**Comment by nanos** [comment]
I have had a look through my Fedifetcher logs to see what servers are reported as showing up in my feed, but not supported by FediFetcher.

Here is a breakdown:

**Mastodon compatible as they are** (implemented in #238)
 - [Fedibird](https://fedidb.com/software/fedibird): Mastodon fork
 - [Kmyblue](https://fedidb.com/software/kmyblue): Mastodon fork
 - [Cherrypick](https://fedidb.com/software/cherrypick): Misskey fork
  
 **Mastodon compatible with very small changes** (implemented in #238)
 - [Mitra](https://fedidb.com/software/mitra): Another microblogging solution, with a slightly more Pleroma like API flavour.

**Lemmy-like**
- [Piefed](https://fedidb.com/software/piefed): Should mostly be understood by the Lemmy client, with some minor changes. I will look into this.

**Not possible for FediFetcher**
The following apps are not possible for FediFetcher to support:
 - GoToSocial and Friendica: by far the most common, but already ruled out, as they require auth for both context and post fetching
 - Hollo, snac, NeoDB: tiny, and they require auth for both context and post fetching
 - Meisskey: could probably fetch context, but there doesn't appear to be a public API to fetch missing posts.
